commands/push.go

```diff
@@ -141,3 +150,30 @@ func validateImages(functions map[string]stack.Function) []string {
 	}
 	return invalidImages
 }
+
+func overrideImagePrefix(yamlImageName string) string {
+	if yamlImageName == "" {
+		return ""
+	}
+
+	imageNameParts := strings.Split(yamlImageName, "/")
```

Comment:
This should be done without using split. There are other string methods available

Reply:
It might be worth looking at the docker distribution repo, they have a package dedicated to  parsing docker image names.  Many of the public objects aren't going to be useful, but the regex could be helpful or the logic in some of the parsers. 

Importantly, I think this current approach will have issues if the image name contains a domain, see the grammar section for https://github.com/docker/distribution/blob/6d62eb1d4a3515399431b713fde3ce5a9b40e8d5/reference/reference.go#L7

Adding a test case fo something like `quay.io/repo/func` would be good

Reply:
Good call, thanks. I'll change it to use the package you linked for (un)marshalling